gk/fib.c

```diff
@@ -872,11 +958,21 @@ ether_cache_put(struct gk_fib *neigh_fib,
 	}
 
 	if (addr->proto == ETHER_TYPE_IPv4) {
+		ret = put_arp((struct in_addr *)
+			&addr->ip.v4, gk_conf->lcores[0]);
+		if (ret < 0)
+			return ret;
+
```

Comment:
I can't comment on the line below, so I copied it here:
```
return neigh_del_ether_cache(
  			&neighbor_fib->u.neigh, &addr->ip.v4.s_addr);
```

If `neigh_del_ether_cache()` fails, we enter an inconsistency state: `put_arp()` released the ether cache entry, but the entry is still being referred by the neighbor table. While trying to recover from this likely impossible failure would be way too much trouble, we should at least log a serious message that this condition happened before returning the return of `neigh_del_ether_cache()`, so this problem can be "easily" diagnosticated in production (if it ever happens).

ATTENTION: The same case for the IPv6 case below.